apps/angular/src/app/features/anime-dashboard/anime-dashboard.component.ts

```diff
@@ -11,12 +24,112 @@ import { AnimeTableComponent } from './components/anime-table/anime-table.compon
 	templateUrl: './anime-dashboard.component.html',
 	styleUrls: ['./anime-dashboard.component.css'],
 	standalone: true,
-	imports: [CommonModule, AnimeTableComponent],
+	imports: [
+		CommonModule,
+		AnimeTableComponent,
+		MatPaginator,
+		BasicProgressSpinnerComponent,
+		MatFormFieldModule,
+		MatInputModule,
+		FormsModule,
+		MatSelectModule,
+		MatButtonModule,
+	],
 	changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class AnimeDashboardComponent {
+export class AnimeDashboardComponent implements OnInit {
 	private readonly animeService: AnimeService = inject(AnimeService);
 
+	/** Current route. */
+	protected route: ActivatedRoute = inject(ActivatedRoute);
+
 	/** Anime list. */
-	protected readonly animeList$ = this.animeService.getAnimeList();
+	protected animeList$: Observable<readonly Anime[]> | null = null;
+
+	/** Current page index. */
+	protected currentPage$ = this.animeService.observableAnimeParams$.pipe(
+		map(params => params.pageIndex),
+	);
+
+	/** Maximum number of items per page. */
+	protected pageSize$ = this.animeService.observableAnimeParams$.pipe(
+		map(params => params.pageSize),
+	);
+
+	/** Loading state from the service. */
+	protected isLoading$ = this.animeService.observableIsLoading$;
+
+	/** Anime count. */
+	protected animeTotal = AnimeParams.defaultValues.animeTotal;
+
+	/** Search input value. */
+	protected searchValue = AnimeParams.defaultValues.searchValue;
+
+	/** Anime type values. */
+	protected animeTypes = Object.values(AnimeType);
```

Comment:
```suggestion
	protected readonly isLoading$ = this.animeService.observableIsLoading$;

	/** Anime count. */
	protected readonly animeTotal = AnimeParams.defaultValues.animeTotal;

	/** Search input value. */
	protected readonly searchValue = AnimeParams.defaultValues.searchValue;

	/** Anime type values. */
	protected readonly animeTypes = Object.values(AnimeType);
```

We follow the immutability approach; most often, class fields are immutable. Let's make decisions that will be based on this principle

Reply:
Hope i get it right :crossed_fingers: